docs/docs/reference/production-deployment.md

```diff
@@ -0,0 +1,97 @@
+
+
+### Authenticate
+
+You should provide a TCP or HTTP(s) load balancer between end users and the Authorize service.
+
+Authenticate is compatible with any L4 or L7/HTTP load balancer.  Session stickiness should not be required and it is typical to have Authenticate be a named vhost on the same L7 load balancer as the Proxy service.
+
+### Authorize
+
+You do **not** need to provide a load balancer between the Proxy and the Authorize service.  Authorize utilizes GRPC and thus has special requirements if you should chose to use an external load balancer.  GRPC can perform client based load balancing, and in most configurations is the best architecture.
+
+By default, the Proxy will automatically connect to all IPs returned by a DNS query for the name of the Authorize serivce.  Proxy will then regularly re-query DNS for changes to the Authorize service cluster.  Health checks and failover are automatic.
+
+If you chose to use an external proxying load balancer instead of the default client implementation, it must be able to understand HTTP2 to properly distribute requests.  In L4 load balancing configurations, GRPC/HTTP2 traffic will be pinned to a single Authorize instance, due to the way HTTP2 multiplexes requests over a single established connection.  
+
+## High Availability
+
+As mentioned in [scaling](#scaling), Pomerium components themselves are stateless and support horizontal scale out for both availability and performance reasons.
+
+A given service type does not require communication with its peer instances to provide high availability.  Eg, a proxy service does not communicate with other proxies.
+
+Regardless of the service mode, it is recommended you run 2+ instances of Pomerium with as much physical and logical separation as possible.  For example, in Cloud environments, you should deploy instances of each service to at least 2 different zones.  On-prem environments should deploy 2+ instances to independent hardware.  
+
+Ensure that you have enough spare capacity to handle the scope of your failure domains.
+
+## SSL Certificates
+
+Pomerium utilizes SSL end to end, so the placement, certificate authorities and covered subjects are critical to align correctly.  
+
+In a typical deployment two certs are required:
+
+- A wildcard certificate which covers the external `from` names, the Proxy service's external name and the Authenticate service's external name
+  - Must be trusted by user browsers or clients
+  - Must cover subject names from the user perspective
+- A certificate which covers the Authorize service's name
+  - Must be trusted by the Proxy
+  - Must cover the subject name from the Proxy's perspective
+
+If you have SSL enabled applications behind the proxy:
+  - you may provide a client certificate for the Proxy service to connect to upstreams with and verify
```

Comment:
upstreams?

Reply:
Borrowing from nginx nomenclature.  I actually went back and forth on this.  If you think downstream is better, I'm good with that.  It's probably clearer.